src/Hudl.Mjolnir/Events/GaugeLogMetrics.cs

```diff
@@ -0,0 +1,115 @@
+        }
+
+        public void BreakerFixed(string breakerName)
+        {
+            var log = $"BreakerFixed - [Breaker={breakerName}]";
+            _diagnosticLog.Debug(log);
+            _breakerTrippedLog.Debug(log);
+        }
+
+        public void BreakerGauge(string breakerName, long configuredMinimumOperations, long configuredWindowMillis, int configuredThresholdPercent, long configuredTrippedDurationMillis, bool configuredForceTripped, bool configuredForceFixed, bool isTripped, long windowSuccessCount, long windowFailureCount)
+        {
+            var log = $"BreakerGauge - [Breaker={breakerName}, ConfiguredMinimumOperations={configuredMinimumOperations}, ConfiguredWindowMs={configuredWindowMillis}, ConfiguredThresholdPercent={configuredThresholdPercent}, ConfiguredTrippedDurationMs={configuredTrippedDurationMillis}, ConfiguredForceTripped={configuredForceTripped}, ConfiguredForcedFixed={configuredForceFixed}, IsTripped={isTripped}, WindowSuccessCount={windowSuccessCount}, WindowFailureCount={windowFailureCount}]";
+            _diagnosticLog.Debug(log);
+            _breakerGaugeLog.Debug(log);
+            if (isTripped)
+            {
+                _breakerTrippedLog.Debug(log);
+            }
+        }
+
+        public void BreakerSuccessCount(string breakerName, string commandName)
+        {
+            _diagnosticLog.Debug($"BreakerSuccessCount - [Breaker={breakerName}, Command={commandName}]");
+        }
+
+        public void BreakerTripped(string breakerName)
+        {
+            var log = $"BreakerTripped - [Breaker={breakerName}]";
+            _diagnosticLog.Debug(log);
+            _breakerTrippedLog.Debug(log);
+        }
+
+        public void BulkheadGauge(string bulkheadName, string bulkheadType, int maxConcurrent, int countAvailable)
+        {
+            var gaugeLog = $"BulkheadGauge - [Bulkhead={bulkheadName}, BulkheadType={bulkheadType}, MaxConcurrent={maxConcurrent}, CountAvailable={countAvailable}]";
+            _diagnosticLog.Debug(gaugeLog);
+            _bulkheadGaugeLog.Debug(gaugeLog);
+            if (countAvailable == 0)
+            {
+                _concurrencyExceededLog.Debug(gaugeLog);
+            }
+            // Log the current rejections that occurred since the last gauge
+            if (_currentBulkheadsRejected.TryGetValue(bulkheadName, out int currentCount))
+            {
+                if (currentCount > 0)
+                {
+                    _concurrencyExceededLog.Debug($"BulkheadRejections since last gauge - [Bulkhead={bulkheadName}, BulkheadType={bulkheadType}, MaxConcurrent={maxConcurrent}, CountAvailable={countAvailable}, Rejections={currentCount}]");
+                    // Remove the rejections we've just logged from the current rejection count
+                    _currentBulkheadsRejected.AddOrUpdate(bulkheadName, 0, (b, c) => c - currentCount);
```

Comment:
Might be able to get rid of some of the code here that calculates deltas, and instead just keep a running incrementing count of rejections. The logging aggregator/search (e.g. Sumo Logic, Splunk) could possibly be used to find the per-window counts, depending on its capabilities.